src/Components/WebAssembly/DebugProxy/src/Program.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.DebugProxy
+{
+    public class Program
+    {
+        static int Main(string[] args)
+        {
+            var app = new CommandLineApplication(throwOnUnexpectedArg: false)
+            {
+                Name = "webassembly-debugproxy"
+            };
+            app.HelpOption("-?|-h|--help");
+
+            var browserHostOption = new CommandOption("-b|--browser-host", CommandOptionType.SingleValue)
+            {
+                Description = "Host on which the browser is listening for debug connections. Example: http://localhost:9300"
+            };
+
+            var ownerPidOption = new CommandOption("-op|--owner-pid", CommandOptionType.SingleValue)
+            {
+                Description = "ID of the owner process. The debug proxy will shut down if this process exits."
+            };
+
+            app.Options.Add(browserHostOption);
+            app.Options.Add(ownerPidOption);
+
+            app.OnExecute(() =>
+            {
+                var host = Host.CreateDefaultBuilder(args)
+                    .ConfigureAppConfiguration((hostingContext, config) =>
+                    {
+                        config.AddCommandLine(args);
+                    })
+                    .ConfigureWebHostDefaults(webBuilder =>
+                    {
+                        webBuilder.UseStartup<Startup>();
+
+                        // By default we bind to a dyamic port
+                        // This can be overridden using an option like "--urls http://localhost:9500"
+                        webBuilder.UseUrls($"http://127.0.0.1:0");
+                    })
+                    .ConfigureServices(serviceCollection =>
+                    {
+                        serviceCollection.AddSingleton(new DebugProxyOptions
+                        {
+                            BrowserHost = browserHostOption.HasValue()
+                                ? browserHostOption.Value()
+                                : "http://127.0.0.1:9222",
+                        });
+                    })
+                    .Build();
+
+                if (ownerPidOption.HasValue())
+                {
+                    var ownerProcess = Process.GetProcessById(int.Parse(ownerPidOption.Value()));
+                    _ = ExitWhenOwnerProcessExits(ownerProcess, host);
+                }
+
+                host.Run();
+
+                return 0;
+            });
+
+            try
+            {
+                return app.Execute(args);
+            }
+            catch (CommandParsingException cex)
+            {
+                app.Error.WriteLine(cex.Message);
+                app.ShowHelp();
+                return 1;
+            }
+        }
+
+        private static async Task ExitWhenOwnerProcessExits(Process ownerProcess, IHost host)
+        {
+            while (true)
+            {
+                if (ownerProcess.HasExited)
```

Comment:
Are you able to attach to `Process.Exited` event? I'm not a 100% sure if you're able to set `EnableRaisingEvents` after a process has been started, but that may be better than polling. Eg. https://github.com/dotnet/aspnetcore/blob/b125c76d5f373b8f7653e252e9202ff2b6c2c45d/src/Tools/dotnet-watch/test/AwaitableProcess.cs#L141-L148

Reply:
Good plan. I hadn't noticed the existence of that event. It does seem to work even for existing processes.